### PR TITLE
Bf2513 jump to coordinates

### PR DESCRIPTION
**Pull request by jduncanRadBlue**

### CODE REVIEW

File: app/utils/openlayers.SearchControl.js [hunk 1]
```diff
@@ -2,15 +2,841 @@ import openlayers from 'openlayers'
...
+  }
+
+  _convertGeospatialCoordinateFormat(inputString) {
+
+    var ddPattern = /(\-?\d{1,2}[.]?\d*)[\s+|,?]\s*(\-?\d{1,3}[.]?\d*)/
+    var dmsPattern = /(\d{2})[^\d]*(\d{2})[^\d]*(\d{2}[.]?\d*)([n|N|s|S])[^\w]*(\d{3})[^\d]*(\d{2})[^d]*(\d{2}[.]?\d*)([e|E|w|W])/
+    var mgrsPattern = /(\d{1,2})([a-zA-Z])[^\w]*([a-zA-Z])([a-zA-Z])[^\w]*(\d{5})[^\w]*(\d{5})/
+
+    if (inputString.match(dmsPattern)) {
+      var latitude = this.dmsToDd(RegExp.$1, RegExp.$2, RegExp.$3, RegExp.$4)
```
Comment (dbazile) [inline app/utils/openlayers.SearchControl.js]: I don't think we want to use `RegEx.$1`.

I have honestly never seen this before and MDN also recommends against their use:
screen shot 2016-06-09 at 12 44 26 pm

File: app/utils/openlayers.SearchControl.js [hunk 2]
```diff
@@ -2,15 +2,841 @@ import openlayers from 'openlayers'
...
+      var location = this.mgrsToDd(RegExp.$1, RegExp.$2, RegExp.$3, RegExp.$4, RegExp.$5, RegExp.$6)
+
+
+      return this._convertGeospatialCoordinateFormat(location)
+    }
+    else { return false }
   }
+
+
+    ddToDms (dd, position) {
```
Comment (dbazile) [inline app/utils/openlayers.SearchControl.js]: Can we remove these conversion functions from the class and just leave them as top-level functions?  Most if not all of them don't need references to `this`.

Comment (dbazile) [inline app/utils/openlayers.SearchControl.js]: Scratch that.  Now that I've had a better look, these conversion utilities seem pretty crusty.  I think we might be better served using the coordinates thing I sent you the other day:
https://gist.github.com/dbazile/8a328e139fa824f2634016db2515dafc

It needs to be cleaned up and repackaged but I believe it's a cleaner solution than this one.

File: app/utils/openlayers.SearchControl.js [hunk 3]
```diff
@@ -2,15 +2,841 @@ import openlayers from 'openlayers'
 
 export default class SearchControl extends openlayers.control.Control {
   constructor(className) {
+
     const element = document.createElement('div')
     super({element})
     element.className = `${className || ''} ol-unselectable ol-control`
     element.title = 'Jump to a specific location or coordinate'
+    element.id = 'searchCoordBtn'
```
Comment (dbazile) [inline app/utils/openlayers.SearchControl.js]: Can we remove this ID?  That way we can avoid a collision if the control is ever reused.

File: app/utils/openlayers.SearchControl.js [hunk 4]
```diff
@@ -2,15 +2,841 @@ import openlayers from 'openlayers'
...
+    }
+    return this._dialog
+  }
+
+  _formSubmitted(event){
+
+    console.log(this._dialog.querySelector("input").value)
+    event.preventDefault()
+
+    var [x,y] = this._convertGeospatialCoordinateFormat(this._dialog.querySelector("input").value)
```
Comment (dbazile) [inline app/utils/openlayers.SearchControl.js]: We need more robust error handling here.  For example, if the coordinate doesn't parse, the script explodes at this point.

File: app/utils/openlayers.SearchControl.js [hunk 5]
```diff
@@ -2,15 +2,841 @@ import openlayers from 'openlayers'
...
+
+    var view = map.getView()
+    view.setCenter(openlayers.proj.transform([long, lat], 'EPSG:4326', 'EPSG:3857'))
+    view.setZoom(8)
+    this._dialog.style.display = 'none'
+    this._dialog.querySelector("input").value = ""
+  }
+
+  _searchClicked() {
+    this.getDialog().style.display = 'block'
```
Comment (dbazile) [inline app/utils/openlayers.SearchControl.js]: This should probably toggle so the user has some means of cancelation if they don't want to jump.

File: app/utils/openlayers.SearchControl.js (hunk 4 above)
Comment (jduncanRadBlue) [inline app/utils/openlayers.SearchControl.js]: Will add error handling to the new conversion utility

**Pull request merged**